pkgs/servers/monitoring/prometheus/shelly-exporter.nix

```diff
@@ -0,0 +1,24 @@
+{ lib, buildGoModule, fetchFromGitHub, nixosTests }:
+
+buildGoModule rec {
+  pname = "shelly_exporter";
+  version = "1.0.0";
+
+  src = fetchFromGitHub {
+    owner = "aexel90";
+    repo = pname;
+    rev = "v" + version;
+    sha256 = "sha256-L0TuBDq5eEahQvzqd1WuvmXuQbbblCM+Nvj15IybnVo=";
+  };
+
+  vendorSha256 = "sha256-BCrge2xLT4b4wpYA+zcsH64a/nfV8+HeZF7L49p2gEw=";
+
+  passthru.tests = { inherit (nixosTests.prometheus-exporters) shelly_exporter; };
```

Comment:
The test seems to be missing.

Reply:
What should I add then?

Reply:
You need to your test in `nixos/tests/prometheus-exporters.nix`.

For our purposes I'm guessing something simple would be good enough e.g. a JSON file containing `{}` for the metrics file and checking the endpoint can be accessed could be good enough. We would need to access the Shelly Cloud to do something more precise.

Reply:
It's not possible to run the `shell_exporter` without a valid JSON file... so it's impossible to do a valid test.

What should we do then?

Reply:
It looks like we can:

```
$ ./result/bin/shelly_exporter -metrics-file=/tmp/test.json
metrics available at http://127.0.0.1:9784/metrics
$ curl http://127.0.0.1:9784/metrics
[...]
$ cat /tmp/test.json
{}
```

Reply:
Fixed now.